tests/utils.py

```diff
@@ -275,6 +279,35 @@ def invocation_argv():
     sys.argv[:] = saved_argv
 
 
+# Workaround for tornado bug #1573, already fixed in master, but not yet
+# available. Remove when upgrading tornado.
+def bind_unused_port(reuse_port=False):
+    """Binds a server socket to an available port on localhost.
+
+    Returns a tuple (socket, port).
+    """
+    sock = tornado.netutil.bind_sockets(None,
+                                        '127.0.0.1',
+                                        family=socket.AF_INET,
+                                        reuse_port=reuse_port)[0]
+    port = sock.getsockname()[1]
+    return sock, port
+
+
+class AsyncHTTPTestCase(tornado.testing.AsyncHTTPTestCase):
+    """Base class workaround for the above condition."""
+    def setUp(self):
```

Comment:
Maybe we don't want to overload `setUp` here given we frequently subclass `TempMixin` and use the `TempMixin.setUp` instead.  How about making this a module-level function that takes a test case as an argument?